### PR TITLE
Enable promlint in metrics tests

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Enable lint check by default in unit tests.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/priority important-soon
/assign @logicalhan

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/testutil/promlint.go [hunk 1]
```diff
@@ -101,3 +123,42 @@ func NewPromLinter(r io.Reader) *Linter {
...
+	for index := range problems {
+		problemsMsg = append(problemsMsg, problems[index].String())
+	}
+
+	return strings.Join(problemsMsg, ",")
+}
+
+// shouldIgnore returns true if metric in the exception list, otherwise returns false.
+func shouldIgnore(metricName string) bool {
+	for i := range exceptionMetrics {
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply staging/src/k8s.io/component-base/metrics/testutil/promlint.go:139]: I've thought about it. The exception list originally organized by components, some metrics with the same name may exist in more than one component.

It's worth to reconsider it. Maybe map[string]struct{}.

[the remark this answers is not in the document]
Comment (logicalhan) [reply staging/src/k8s.io/component-base/metrics/testutil/promlint.go:139]: You'll have collisions either way, no?

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**